src/client.rs

```diff
@@ -239,126 +236,85 @@ impl Tidal {
+        self.playlists().search(term, limit).await
     }
 
     pub async fn user_playlists(&self) -> ClientResult<Vec<Playlist>> {
-        let user_id = self.user_id();
-        let url = format!("/users/{}/playlists", user_id);
-        let result = self.get(&url, &mut HashMap::new()).await?;
-        let playlists = Self::convert_result::<TidalItems<Playlist>>(&result)?.items;
-        Ok(playlists)
+        warn!("DEPRECATION WARNING!: This method will be deprecated in the next version. Please favor using .playlists().user_playlists()");
+        self.playlists().user_playlists().await
     }
 
     pub async fn playlist_tracks(&self, id: &str) -> ClientResult<Vec<Track>> {
-        let url = format!("/playlists/{}/tracks", id);
-        let result = self.get(&url, &mut HashMap::new()).await?;
-        let tracks = Self::convert_result::<TidalItems<Track>>(&result)?.items;
-        Ok(tracks)
+        warn!("DEPRECATION WARNING!: This method will be deprecated in the next version. Please favor using .playlists().tracks()");
+        self.playlists().tracks(id).await
     }
 
-    pub async fn playlist_add_tracks(
-        &self,
-        id: &str,
-        tracks: Vec<Track>,
-        add_dupes: bool,
-    ) -> ClientResult<Playlist> {
-        let url = format!("/playlists/{}/items", id);
-        let etag: String = self.etag(&url).await?;
-        let track_ids: Vec<String> = tracks
-            .iter()
-            .map(|track| track.id.unwrap().to_string())
-            .collect();
-        let track_ids: String = track_ids.join(",");
-
-        let on_dupes: String = if add_dupes {
-            "ADD".to_owned()
-        } else {
-            "FAIL".to_owned()
-        };
-
-        let mut form: HashMap<&str, &str> = HashMap::new();
-        form.insert("trackIds", &track_ids);
-        form.insert("onDupes", &on_dupes);
+    pub async fn playlist_add_tracks(&self, id: &str, tracks: Vec<Track>, add_dupes: bool) -> ClientResult<Playlist> {
+        warn!("DEPRECATION WARNING!: This method will be deprecated in the next version. Please favor using .playlists().add_tracks()");
+        self.playlists().add_tracks(id, tracks, add_dupes).await
+    }
 
-        self.post(&url, &form, etag).await?;
-        self.playlist(id).await
+    pub async fn create_playlist(&self, title: &str, description: &str) -> ClientResult<Playlist> {
+        warn!("DEPRECATION WARNING!: This method will be deprecated in the next version. Please favor using .playlists().create()");
+        self.playlists().create(title, description).await
     }
 
-    fn convert_result<'a, T: Deserialize<'a>>(input: &'a str) -> ClientResult<T> {
+    pub fn convert_result<'a, T: Deserialize<'a>>(input: &'a str) -> ClientResult<T> {
         serde_json::from_str::<T>(input).map_err(Into::into)
     }
 }
 
 #[cfg(test)]
-mod tests {
+pub mod tests {
```

Comment:
Why does this test module need to public?

Reply:
Also the methods 🤔

Reply:
@jpalvarezl to reuse `mock_request_success_from_file` and `client` from the `Tidal` test on the `endpoint` tests. I guess they could be better on some sort of "test utils" or something like that, but seemed overkill for the time being. 🤷‍♂️